tests/functional/kata-agent-apis/gha-run.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2024 Microsoft Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+kata_tarball_dir="${2:-kata-artifacts}"
+kata_agent_apis_dir="$(dirname "$(readlink -f "$0")")"
+source "${kata_agent_apis_dir}/../../common.bash"
+
+function install_dependencies() {
+	info "Installing dependencies needed for testing individual agent apis using agent-ctl"
+
+	# Dependency list of projects that we can rely on the system packages
+	# - jq
+	declare -a deps=(
+		jq
+	)
+
+	sudo apt-get update
+	sudo apt-get -y install "${deps[@]}"
+}
+
+function run() {
+	info "Testing agent apis with agent-ctl."
+
+	bash -c ${kata_agent_apis_dir}/run-agent-api-tests.sh
```

Comment:
So, this cannot be tested as part of this PR, due to the `pull_request_target` that we use.
So, I'd suggest adding everything as you did, but just returning 0 here instead of running the tests as part of this PR.

Then, after this one gets merged, you'll open a second PR which will then enable the tests.
WDYT?

Reply:
Sure that makes sense. I will do that.

Reply:
@fidencio I have done the changes.